gufe/tests/storage/test_storagemanager.py

```diff
@@ -0,0 +1,248 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+import pathlib
+import tempfile
+from unittest.mock import MagicMock, patch
+
+import pytest
+from pluggy import _manager
+
+from gufe.storage.externalresource import FileStorage, MemoryStorage
+from gufe.storage.externalresource.base import ExternalStorage
+from gufe.storage.storagemanager import StorageManager
+
+
+@pytest.fixture
+def tmp_scratch_path():
+    """Create a temporary scratch directory."""
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        yield pathlib.Path(tmp_dir)
+
+
+@pytest.fixture
+def memory_storage():
+    """Create a MemoryStorage instance for testing."""
+    return MemoryStorage()
+
+
+@pytest.fixture
+def file_storage(tmp_path):
+    """Create a FileStorage instance for testing."""
+    return FileStorage(tmp_path)
+
+
+@pytest.fixture
+def storage_manager(tmp_scratch_path, memory_storage):
+    """Create a StorageManager instance with MemoryStorage."""
+    return StorageManager(tmp_scratch_path, memory_storage, dag_label="MEM", unit_label="1")
+
+
+@pytest.fixture
+def storage_manager_file_storage(tmp_scratch_path, file_storage):
+    """Create a StorageManager instance with FileStorage."""
+    return StorageManager(tmp_scratch_path, file_storage, dag_label="FILE", unit_label="1")
+
+
+class TestStorageManager:
+    """Test the StorageManager class."""
+
+    def test_init(self, tmp_scratch_path, memory_storage):
+        """Test StorageManager initialization."""
+        manager = StorageManager(tmp_scratch_path, memory_storage, dag_label="MEM", unit_label="1")
+
+        assert manager.scratch_path == tmp_scratch_path
+        assert manager.storage == memory_storage
+        assert isinstance(manager.registry, set)
+        assert len(manager.registry) == 0
+
```

Comment:
```suggestion
assert manager.namespace == "MEM/1"
```

any reason not to check this?